capstone/src/main/java/com/google/step/servlets/RestaurantsServlet.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.step.data.Restaurant;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for getting restaurants from Datastore. */
+@WebServlet("/restaurants")
+public class RestaurantsServlet extends HttpServlet {
+  /**
+   * Returns a list of Restaurants.
+   */
+  // TODO: Change this to be connected to search functionality.
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("RestaurantInfo").addSort("score", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Restaurant> restaurants = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long restaurantKey = (Long) entity.getProperty("restaurantKey");
```

Comment:
This is a big block of repeated code. Consider pulling out a helper method to parse this.

One good way you could do it is to add a static factory method (called `fromEntity`, say) to Restaurants, which returns a Restaurant it constructs from the entity (and throws an exception if it's the wrong kind of entity, etc.)

You can also then add a unit test to Restaurant and make sure it works correctly.

Reply:
Added the factory method and a couple of unit tests for it.